File: src/lib/software/godot.py
from src.lib.software.variants.godot_cs import GodotCS
from src.lib.software import BaseSoftware, SoftwareCategory
from src.lib.software.variants.godot_stable import GodotStable

class Godot(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.key = 'godot'
        self.name = 'Godot'
        self.category = [SoftwareCategory.Creative, SoftwareCategory.GameDevelopment]
        self.variants = [GodotStable(), GodotCS()]
        self.icon = 'godot.png'
        self.homepage = 'https://godotengine.org'

    def resolve_download_url(self):
        pass

File: packages/app/src/services/software/definitions/godot.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'godot';
	public name = 'Godot';
	public category = [SoftwareCategory.Creative, SoftwareCategory.GameDevelopment];
	public downloadName = '';
	public icon = 'godot.png';
	public homepage = 'https://godotengine.org';

	public async resolveDownloadUrl() {
		return ok('');
	}
}
